src/ipahealthcheck/ipa/certs.py

```diff
@@ -724,6 +724,81 @@ def check(self):
                         self, constants.SUCCESS, key=cert)
 
 
+def check_agent(plugin, base_dn):
```

Comment:
If it took a string you could differentiate agent type:

e.g check_agent(plugin, base_dn, type):
...
logger.debug('%s agent descripiton should be %s', type, description)
...
        yield Result(plugin, constants.ERROR,
                     description=description,
                     msg='%s agent not found in LDAP' % type)

I guess it's implicit so this isn't a hard ask but it does read better IMHO, particularly in the human output.